### DISCUSSION TITLE
希望可以直接通过容器名称代理

### DISCUSSION THREAD

**Discussion by feewg** [1Panel-dev/1Panel#6115] [Q&A]
### 1Panel 版本

v1.10.13-lts

### 请描述您的需求或者改进建议

- [ ] 在创建应用代理时，可以直接通过容器名称+端口代理，不需要将容器端口映射到本地，也不需要去查找容器ip

**Comment by zhengkunwang223** [comment]
当前 Openresty 是 host 模式 代理容器时 填写 127.0.0.1:容器端口即可

**Comment by feewg** [comment]
> 当前 Openresty 是 host 模式 代理容器时 填写 127.0.0.1:容器端口即可

这里的意思是我不想将容器的端口映射出来，希望直接通过docker网络访问容器内部端口并反代容器

**Comment by zhengkunwang223** [comment]
> > 当前 Openresty 是 host 模式 代理容器时 填写 127.0.0.1:容器端口即可
> 
> 这里的意思是我不想将容器的端口映射出来，希望直接通过docker网络访问容器内部端口并反代容器

那你可以把 OpenResty 的 host 模式删除
删除 docker-compose 文件中的    network_mode: host
增加       
networks:
      - 1panel-network

networks:
  1panel-network:
    external: true

或者你容器的 Docker 网络

**Comment by feewg** [comment]
> > > 当前 Openresty 是 host 模式 代理容器时 填写 127.0.0.1:容器端口即可
> > 
> > 
> > 这里的意思是我不想将容器的端口映射出来，希望直接通过docker网络访问容器内部端口并反代容器
> 
> 那你可以把 OpenResty 的 host 模式删除 删除 docker-compose 文件中的 network_mode: host 增加 networks: - 1panel-network
> 
> networks: 1panel-network: external: true
> 
> 或者你容器的 Docker 网络


OpenResty的host模式是不需要更改的，需要的是在反代的时候能够通过容器名称反代，不需要将容器的内部端口映射出来

**Comment by zhengkunwang223** [comment]
> > > > 当前 Openresty 是 host 模式 代理容器时 填写 127.0.0.1:容器端口即可
> > > 
> > > 
> > > 这里的意思是我不想将容器的端口映射出来，希望直接通过docker网络访问容器内部端口并反代容器
> > 
> > 
> > 那你可以把 OpenResty 的 host 模式删除 删除 docker-compose 文件中的 network_mode: host 增加 networks: - 1panel-network
> > networks: 1panel-network: external: true
> > 或者你容器的 Docker 网络
> 
> OpenResty的host模式是不需要更改的，需要的是在反代的时候能够通过容器名称反代，不需要将容器的内部端口映射出来

host 模式下通过容器名称反代 需要解决容器名称的解析问题 你有解决方案吗？
